src/main/java/org/opensearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -113,150 +128,215 @@ protected AbstractApiAction(
-                default:
-                    throw new IllegalArgumentException(request.method() + " not supported");
-            }
-        } catch (JsonMappingException jme) {
-            throw jme;
-            // TODO strip source
-            // if(jme.getLocation() == null || jme.getLocation().getSourceRef() == null) {
-            // throw jme;
-            // } else throw new JsonMappingException(null, jme.getMessage());
-        }
+    protected final ValidationResult<SecurityConfiguration> processPutRequest(final RestRequest request) throws IOException {
+        return withRequiredResourceName(request).map(name -> loadSecurityConfigurationWithRequestContent(name, request))
+            .map(this::resourceCanBeChanged);
     }
 
-    protected void requestContentInvalid(final RestRequest request, final RestChannel channel, final ToXContent toXContent) {
-        request.params().clear();
-        badRequestResponse(channel, toXContent);
+    final void saveOrUpdateConfiguration(
+        final Client client,
+        final SecurityDynamicConfiguration<?> configuration,
+        final OnSucessActionListener<IndexResponse> onSucessActionListener
+    ) {
+        saveAndUpdateConfigs(this.securityIndexName, client, getConfigName(), configuration, onSucessActionListener);
     }
 
-    protected void handleDelete(final RestChannel channel, final RestRequest request, final Client client, final JsonNode content)
-        throws IOException {
+    protected final String nameParam(final RestRequest request) {
         final String name = request.param("name");
-
-        if (name == null || name.length() == 0) {
-            badRequestResponse(channel, "No " + getResourceName() + " specified.");
-            return;
+        if (Strings.isNullOrEmpty(name)) {
+            return null;
         }
+        return name;
+    }
 
-        final SecurityDynamicConfiguration<?> existingConfiguration = load(getConfigName(), false);
-
-        if (!isWriteable(channel, existingConfiguration, name)) {
-            return;
+    protected final ValidationResult<String> withRequiredResourceName(final RestRequest request) {
+        final var resourceName = nameParam(request);
+        if (resourceName == null) {
+            return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("No " + getResourceName() + " specified."));
         }
+        return ValidationResult.success(resourceName);
+    }
 
-        boolean existed = existingConfiguration.exists(name);
-        existingConfiguration.remove(name);
-
-        if (existed) {
-            AbstractApiAction.saveAndUpdateConfigs(
-                this.securityIndexName,
-                client,
-                getConfigName(),
-                existingConfiguration,
-                new OnSucessActionListener<IndexResponse>(channel) {
+    protected final ValidationResult<SecurityConfiguration> loadFilteredConfiguration(final String resourceName) throws IOException {
```

Comment:
What does filtered mean here exactly? If I'm reading this correctly it looks like its removing hidden, clearing hashes and setting meta to null?

In some instances I know `filter` is used to search through the list of roles for roles that match a given name. Is this method used for that? Example: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java#L186-L188

Reply:
> What does filtered mean here exactly? If I'm reading this correctly it looks like its removing hidden, clearing hashes and setting meta to null?
> 

Yes it is and I think the name is misleading. What do you think to call it something like:
`loadWithoutSensitiveData` or something else. Im not good in naming :-)

Reply:
Done

Reply:
done